gridtools/shared/mosaicobj.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional, Dict
+from dataclasses import dataclass,field
+import sys
+import os
+import xarray as xr
+import numpy as np
+import numpy.typing as npt
+
+from gridtools_lib import GridObj
+
+@dataclass
+class MosaicObj:
+    mosaic_file: str = None
+    output_file: str = None
+    ntiles: int = None
+    ncontact: int = None
+    mosaic_name: str = None
+    gridlocation: str = None
+    gridfiles: npt.NDArray[np.str_] = None
+    gridtiles: npt.NDArray[np.str_] = None
+    contacts: npt.NDArray[np.str_] = None
+    contact_index: npt.NDArray[np.str_] = None
+    mosaic: object = field(init=False)
+    grid_dict: Optional[Dict] | None = field(default_factory=dict)
+
+    def __post_init__(self):
+        if self.mosaic_file is not None and self.gridfiles is None:
+            try:
+                self.dataset = xr.open_dataset(self.mosaic_file)
+            except FileNotFoundError:
+                sys.exit("No such file or directory: {}".format(self.mosaic_file))
+            self.gridfiles = self.get_gridfiles()
+
+    def get_gridfiles(self):
+        try:
+            files = list(self.dataset['gridfiles'][:].values.flatten())
+            return [f.decode('utf-8') for f in files]
+        except AttributeError:
+                print("Error: Mosaic file not provided as an attribute, unable to return gridfiles")
+    def get_ntiles(self):
+        try:
+            return  self.dataset['ntiles'].size
+        except AttributeError:
+            print("Error: Mosaic file not provided as an attribute, unable to return number of tiles")
+
+    def file_is_there(self, check_file:str): 
+        if os.path.isfile(check_file): 
+            print(f"File {check_file} exists.") 
+        else:
+            raise FileNotFoundError(f"File {check_file} does not exist.")
+
+    def griddict(self):
+        i = 1
+        for file in self.gridfiles:
+            self.grid_dict[f'tile{i}'] = GridObj.from_file(file)
```

Comment:
we should use the gridtiles name from the mosaic file, perhaps
```
gridtiles = [ self.dataset.values[i].decode('ascii') for i in range(ntiles) ]
self.grid_dict = dict([ (gridtiles[i], GridObj.from_file(self.gridfiles[i]) for i in range(ntiles) ])
```

The explicit for loop is ok too, maybe it makes things easier to read